### PR TITLE
Add loading spinner while tournament bracket is loading / retrieving data

### LINKED ISSUE #9354: Add an indicator for when tournament data is still loading or fails to load from the server
**Describe the new feature:** 

When there are issues on the server side, if a user boots up osu! lazer using `--tournament` while having existing data like user IDs or beatmap IDs stored for lookup, the client will attempt to fetch data and can (partly) fail. 

In the logs you can see that the client fails to connect to the server because it either times out, throws a bad gateaway error or something similar. So, during that time no window shows up yet. 

Because it takes very long (> 10 seconds) for the window to show up a user (or at least me, twice) can be inclined to force close it. 

I'm unable to reproduce this right now for a video, cause when I started to record this the server responded properly again. 

**Proposal designs of the feature:** 
I have a couple of design proposals to show some kind of indicator that the network requests are still ongoing: 

1. A loading indicator that blocks the whole screen before `SetupScreen` shows. This includes a button that the user can press to cancel this process with the consequence that no beatmap or player data that needs to be refreshed is refreshed. 

2. Have a loading indicator somewhere on the `SetupScreen` in an `ActionableInfo` section indicating that data is still trying to load from the server and show any failures. 

Both if these proposals mean that the lookups for users and beatmaps happen while the UI is being constructed rather than before. 

I included these logs, because I first considered it a bug report, to demonstrate a scenario that could need such a thing: 

[network.log](https://github.com/ppy/osu/files/4819074/network.log)
[performance.log](https://github.com/ppy/osu/files/4819075/performance.log)
[runtime.log](https://github.com/ppy/osu/files/4819076/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4819077/updater.log)
[database.log](https://github.com/ppy/osu/files/4819078/database.log)

### PR DESCRIPTION
**Pull request by peppy**
Closes https://github.com/ppy/osu/issues/9354. Not putting too much thought or effort into this because the eventual goal is to move it inside the osu! client itself.

### CODE REVIEW

File: osu.Game.Tournament/TournamentGameBase.cs [hunk 1]
```diff
@@ -127,6 +124,15 @@ private void readBracket()
 
             if (addedInfo)
                 SaveChanges();
+
+            ladder.CurrentMatch.Value = ladder.Matches.FirstOrDefault(p => p.Current.Value);
+
+            Schedule(() =>
+            {
+                dependencies.Cache(ladder);
```
Comment (smoogipoo) [inline osu.Game.Tournament/TournamentGameBase.cs:132]: Are you sure about this? `TournamentGame` derives from this class and adds `TournamentSceneManager` in its BDL which then adds all the scenes that resolve `LadderInfo` in its BDL. It seems like this is going to occur too late?

File: osu.Game.Tournament/TournamentGameBase.cs [hunk 2]
```diff
@@ -70,8 +69,6 @@ private void readBracket()
 
             Ruleset.BindTo(ladder.Ruleset);
```
Comment (smoogipoo) [inline osu.Game.Tournament/TournamentGameBase.cs:70]: Also not sure about this one - we're on an async context after all.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game.Tournament/TournamentGameBase.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Tournament/TournamentGameBase.cs:132]: In-fact it seems `TournamentSceneManager` itself has a dependency. I can't replicate every time but:

`osu.Framework.Allocation.DependencyNotRegisteredException: The type TournamentSceneManager has a dependency on LadderInfo, but the dependency is not registered.`

Comment (peppy) [reply osu.Game.Tournament/TournamentGameBase.cs:132]: Hmm interesting. Will need to move that inside the continuation callback, good catch.

**Comment by peppy** [comment]
Those two issues should now be resolved.

**Review by smoogipoo — APPROVED** [review]
I think this is ok

**Pull request merged**